### PR TITLE
Add export keys menu item.

### PR DESCRIPTION
**Pull request by ftwillms**
- Add an export keys menu item
- Add a ValueError check for other menu items when making a request to the RPC server

### CODE REVIEW

File: MainWindow.py [hunk 1]
```diff
@@ -77,15 +107,17 @@ def on_SaveMenuItem_activate(self, object, data=None):
...
-            dialog = Gtk.MessageDialog(self.window, 0, Gtk.MessageType.INFO,
-                                       Gtk.ButtonsType.OK, "Wallet Saved")
-            dialog.format_secondary_text(
-                "Wallet has been saved successfully.")
-            dialog.run()
-            dialog.destroy()
-        else:
+        try:
+            r = global_variables.wallet_connection.request("save")
+            if not r:
```
Comment (dbileck) [inline MainWindow.py]: Is this check still necessary now that you are handling the exception?

**Comment by CaptainMeatloaf** [comment]
Can you rebase this please so that there are no merge conflicts

**Comment by ftwillms** [comment]
@CaptainMeatloaf rebase is complete

File: MainWindow.py [hunk 2]
```diff
@@ -148,18 +150,56 @@ def on_ResetMenuItem_activate(self, object, data=None):
...
+            source_address = self.builder.get_object("AddressTextBox").get_text()
+            # Capture the secret spend key for this specific address
+            r = global_variables.wallet_connection.request("getSpendKeys", params={'address': source_address})
+            spend_secret_key = r.get('spendSecretKey', 'N/A')
+            dialog = Gtk.MessageDialog(self.window, 0, Gtk.MessageType.INFO,
+                                       Gtk.ButtonsType.OK, "Secret Keys")
+            # Find the widget responsible for the OK button
+            ok_btn = dialog.get_widget_for_response(response_id=Gtk.ResponseType.OK)
+            # Modify the OK button label to Copy
+            ok_btn.set_label("Copy")
```
Comment (CaptainMeatloaf) [inline MainWindow.py]: Perhaps this should read as "Copy and Close" rather than just copy?

Comment (dbileck) [reply MainWindow.py]: Seems like a mouthful. Perhaps would be better having separate buttons?

Comment (ftwillms) [reply MainWindow.py]: Yea, could probably leave the bottom button as OK and add a small button for copy (using that icon github uses when you copy the github repo for example)

Comment (CaptainMeatloaf) [reply MainWindow.py]: That sounds good to me

**Comment by dbileck** [comment]
@ftwillms any interest on completing this PR? If not, I am willing to take over.

**Comment by ftwillms** [comment]
Problem a ran into was trying to add a button to the Dialog object - since I was using the standard dialog box, it seemed like it was just ignoring the custom button I was trying to add to it. The easy fix for this is to change the wording of the OK button to "Copy and Close" but the extended fix is to rework that Dialog object into a custom one.

**Comment by dbileck** [comment]
Sure. Will do.

**Comment by dbileck** [comment]
PR #47 opened. Take a look! :)

**Comment by ftwillms** [comment]
Replaced by https://github.com/turtlecoin/turtle-wallet/pull/47

**Pull request closed without merging**